### DISCUSSION TITLE
The element index "View" button is confusing to authors

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#12718] [Ideas] [answered]
The ability to customize the table columns in Craft 4.3's new "View" menu (https://github.com/craftcms/cms/pull/11915) has been a great addition.

It's a little bit of a mixed bag though, as hiding the active sort options inside the disclosure menu is proving to be a minor UX regression in my experience. 

One example (and a fairly frequent issue) is authors being unaware that a structure section has been sorted by an attribute other than "Structure", and wondering why there are no move handles anymore. Admittedly this occurred from time to time before 4.3 as well, but the fact that there's no longer a dedicated sort button, and no immediate way to tell what the current sort attribute is (or even that the sort attribute has been changed from its default), seems to have exacerbated the issue.

Ideally, I think the current sort options should be apparent in the UI, without having to open up a disclosure menu to see them – like before. But, it would probably also be helpful if the "View" button changed its (visual) appearance whenever any of the options within the View menu is changed from their defaults (i.e. both sort options and the default table columns) – similar to how the filter menu toggle changes state, when there are filters applied. 

Disregarding the "New" button, the "View" button is the only UI element at the top of the element index that _never_ changes its state or appearance based on user input, a detail I think is contributing to some of the confusion.

**Comment by brandonkelly** [comment, marked as answer]
I’ve kindof felt the same way, but we were going to move the sorting options into HUD-type menu either way, to separate the sort attribute and direction inputs for accessibility reasons.

But as you said, this issue existed before 4.3 as well. So I’ve been contemplating introducing an entirely new view button for structured tables, rather than treating “Structure” as a sort option.

Just went ahead and did that for 4.4, and it feels much more intuitive to me. (#12722)

**Reply by mmikkel** [reply]
Whoa. Just tested the beta and having structure as a separate view mode does make a lot of sense! I guess it never really felt completely natural as a sort option, and since the new toggle sits outside of the "View" menu, authors now have that important visual cue that was previously missing, re: structures and sorting.

I'm still not a _huge_ fan of having the remaining table view sort options hidden away in the "View" disclosure menu, but this change is definitely going to help a lot with that structure problem specifically 👏

**Reply by brandonkelly** [reply]
Craft 4.4 is out with that change :)
